src/components/forms/DraftFileUpload.tsx

```diff
@@ -0,0 +1,94 @@
+import {ConfirmationDialog} from '../../components/modals/ConfirmationDialog';
+import {FormFieldTitle} from '../../components/forms/forms';
+import {FileStorageObject} from 'src/types/model';
+
+export type DraftFileUploadProps = {
+  description: string,
+  draftId: string,
+  defaultValue: FileStorageObject,
+  id: string,
+  onAddFile: Function,
+  onDeleteFile: Function,
+  required?: boolean,
+  title: string
+}
+
+export const DraftFileUpload = (props: DraftFileUploadProps) => {
+  const {id, draftId, title, description, onAddFile, onDeleteFile, defaultValue, required} = props;
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [open, setOpen] = useState<boolean>(false);
+  const spinnerRef = useRef<HTMLDivElement>(null);
+  const handleClose = () => {setOpen(false);};
+  const handleDeleteClick = () => {setOpen(true);};
+
+  const handleAddFile = async (event: ChangeEvent<HTMLInputElement>, id: string) => {
+    await onAddFile(event, id);
+  };
+
+  const handleDeleteFile = async () => {
+    await onDeleteFile(draftId, defaultValue?.fileStorageObjectId, id);
+  };
+
+  const deleteButton = (defaultValue) ?
+    <div>
+      <Link
+        style={{marginLeft: '15px'}}
+        id={`${defaultValue.fileStorageObjectId}_delete`}
+        className={'glyphicon glyphicon-trash'}
+        onClick={() => handleDeleteClick()}
+        to={`#`}
+      />
+      <ConfirmationDialog
+        title='Delete Attachment'
+        openState={open}
+        close={handleClose}
+        action={() => {
+          setOpen(false);
+          toggleSpinnerRef(true);
+          handleDeleteFile().then(()=>{
+            toggleSpinnerRef(false);
+            if (inputRef.current) {
+              inputRef.current.value = '';
+            }
+          });
+        }}
+        description={`Are you sure you want to delete the file '${defaultValue.fileName}'?`}
+      />
+    </div>:<div/>;
+
+  const handleUploadButtonClick = () => {
+    inputRef.current?.click();
+  };
+
+  const toggleSpinnerRef = (visible: boolean) => {
+    if (spinnerRef?.current) {
+      if (visible) {
+        spinnerRef.current.style.display = 'inline';
+      } else {
+        spinnerRef.current.style.display = 'none';
+      }
+    }
+  };
+
+  return <div>
+    <FormFieldTitle
+      formId={id}
+      title={title}
+      description={description}
+      required={required}
+    /><input type={'file'} ref={inputRef} style={{ display: 'none' }} onChange={(event)=> {
+      toggleSpinnerRef(true);
+      handleAddFile(event, id).then(()=>{
+        toggleSpinnerRef(false);
+      });
+    }}/>
+    <button disabled={defaultValue != null} onClick={handleUploadButtonClick}>Upload a file</button>
+    {defaultValue && <span>{defaultValue.fileName} {deleteButton}</span>}
+    <div ref={spinnerRef} style={{display:'none', textAlign: 'center', height: '44', width: '180'}}>
+      <img src={loadingIndicator} alt={'Loading'}/>
+    </div>
+  </div>;
```

Comment:
Same here for eslint formatting - The line spacing here makes this hard to read/know which components are nested.

Reply:
Running the linter locally, there are some spacing issues in this file, but the indentation looks correct to me.